src/eureka/S1_detector_processing/s1_meta.py

```diff
@@ -0,0 +1,172 @@
+            elif self.dq_sat_mode == 'defined':
+                self.dq_sat_columns = getattr(self, 'dq_sat_columns')  # Force this to be specified if dq_sat_mode is 'defined'
+
+        # Mask curved traces
+        self.masktrace = getattr(self, 'masktrace', False)
+        if self.masktrace:
+            self.window_len = getattr(self, 'window_len')  # Force this to be specified if masking the trace
+            self.expand_mask = getattr(self, 'expand_mask')  # Force this to be specified if masking the trace
+            self.ignore_low = getattr(self, 'ignore_low') # Force this to be specified if masking the trace
+            self.ignore_hi = getattr(self, 'ignore_hi')  # Force this to be specified if masking the trace
+        elif self.bias_correction:
+            raise AssertionError(f'meta.bias_correction has been set to {self.bias_correction} which requires meta.masktrace=True. Meanwhile, meta.masktrace has been set to False. Please update bias_correction or masktrace in your Stage 1 ECF.')
+
+        # Manual reference pixel correction for NIRSpec PRISM when not subtracting BG
+        self.refpix_corr = getattr(self, 'refpix_corr', False)
+        if self.refpix_corr:
+            self.npix_top = getattr(self, 'npix_top')  # Force this to be specified if refpix_corr is True
+            self.npix_bot = getattr(self, 'npix_bot')  # Force this to be specified if refpix_corr is True
+
+        # Background subtraction
+        self.grouplevel_bg = getattr(self, 'grouplevel_bg', False)
+        if self.grouplevel_bg:
+            self.ncpu = getattr(self, 'ncpu', 4)
+            self.bg_y1 = getattr(self, 'bg_y1')  # Force this to be specified if grouplevel_bg is True
+            self.bg_y2 = getattr(self, 'bg_y2')  # Force this to be specified if grouplevel_bg is True
+            self.bg_deg = getattr(self, 'bg_deg', 0)
+            self.bg_method = getattr(self, 'bg_method', 'mean')  # Options: std (Standard Deviation), median (Median Absolute Deviation), mean (Mean Absolute Deviation)
+            self.p3thresh = getattr(self, 'p3thresh', 5)
+            self.bg_disp = getattr(self, 'bg_disp', False)  # Row-by-row BG subtraction (only useful for NIRCam)
+            self.bg_x1 = getattr(self, 'bg_x1', None)  # Left edge of exclusion region for row-by-row BG subtraction
+            self.bg_x2 = getattr(self, 'bg_x2', None)  # Right edge of exclusion region for row-by-row BG subtraction
+
+        # Diagnostics
+        self.isplots_S1 = getattr(self, 'isplots_S1', 1)
+        self.nplots = getattr(self, 'nplots', 5)
+        self.hide_plots = getattr(self, 'hide_plots', False)
+        self.testing_S1 = getattr(self, 'testing_S1', False)
+        self.verbose = getattr(self, 'verbose', True)
+
+        # Project directory
+        self.topdir = getattr(self, 'topdir')  # Must be provided in the ECF
+
+        # Directories relative to topdir
+        self.inputdir = getattr(self, 'inputdir', 'Uncalibrated')
```

Comment:
I've always used `Uncalibrated` personally, but I really don't feel strongly about the default value and can see how `Stage0` could be nice since it would put it at the top of the directory when sorting alphabetically.